Creating view

konny0311 opened the pull request:
#Why
チャットアプリにチャット画面は必要だから
#What
チャット画面の実装
<img width="1440" alt="chat-space" src="https://user-images.githubusercontent.com/34429021/36295688-78ee9782-132c-11e8-9722-760f6de75857.png">

Commit:
Font awesome gem installed

Commit:
Preparation for font-awesome

Commit:
Memo

Commit:
Html of view/messages/index created for the time being

Need to convert it to haml

Commit:
Amended to Haml

Commit:
Header created

Commit:
Index view amended

Commit:
Haml worked correctly

Commit:
Index.haml amended and messages are related to application.scss

Commit:
Text align centered

Commit:
Header sass created

Commit:
Variable for color created

Commit:
Layout tentatively completed

still need to make variable and function for height and width

Commit:
Variables creation

Commit:
View basic layout done

Commit:
Group view on the left done

Commit:
WIP group name on the top

Commit:
Header-left current_user_name under construction

Commit:
View w/o icons done

Commit:
View w/o icons done

Commit:
Almost done except font-awesome error

Commit:
Got file uploading function

But couldn’t apply it to icon yet

Commit:
Underconstruction of img uploading ban

Commit:
Upload-image function done

What to do is refactoring

Commit:
Before @mixin

Commit:
@mixin for contents_header done

Commit:
View of messages completed

konny0311 commented:
chatspaceのチャット画面を実装しました。レビューお願いします！
<img width="1440" alt="chat-space" src="https://user-images.githubusercontent.com/34429021/36295452-0387201e-132b-11e8-8562-0c12a5123815.png">

hzak2xx commented on Gemfile:
```diff
@@ -36,6 +36,7 @@ gem 'jbuilder', '~> 2.5'
 # gem 'capistrano-rails', group: :development
 gem 'haml-rails'
 gem 'erb2haml'
+gem 'font-awesome-rails'
```
gemfileはこの際コメントアウト等消してすっきりさせましょう！

hzak2xx commented on app/assets/stylesheets/_messages.scss:
```diff
@@ -0,0 +1,239 @@
+$height-of-contents-header: 100px;
+$height-of-footer:          50px;
+// width variables
+$width-of-left:             300px;
```
変数はconfigフォルダを作成し、そちらにファイルを作成して読み込ませましょう

hzak2xx commented on app/assets/stylesheets/_messages.scss:
```diff
@@ -0,0 +1,239 @@
+  width: $width;
+  background-color: $background-color;
+  float: $float;
+}
```
こちらはmixinフォルダを作ってそこから読み込ませましょう

hzak2xx commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,66 @@
+%body
```
bodyはapplication.html.hamlに記述されていると思うので不要です

hzak2xx commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,66 @@
+%body
+  %header
```
こちらは通知の要素でしょうか？でしたら、div要素で作ってクラス名を`notification`などとしましょう。それに対応するcssはまたファイルを作成し、mixinフォルダに入れて読み込んでください

hzak2xx commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,66 @@
+      / chat in a selected group shown
+      .contents_main--body
+        %ul
+          %li
```
`ul`,`li`のちに必要になるのでクラス名つけましょう

hzak2xx commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,66 @@
+          %li
+            member-a
+            %span posted date
+            %p message-1
```
`span`,`p`もクラス名つけましょう。

hzak2xx commented on app/assets/stylesheets/_messages.scss:
```diff
@@ -0,0 +1,239 @@
+          $mainleft-color,
+          left);
+
+            li {
```
なるべくクラス名をつけてcss適用した方がいいです

Commit:
deleting comment-outs in gem file

Commit:
_config.scss created

Variables and mixin in the _message.scss were moved to _config.scss.
And @import method was added in the _message

Commit:
_mixin file created

Commit:
Change "header" into "#notification" and correct indentation

Commit:
Put names of class on elements

Commit:
Create Mixin folder and put _mixin and _notification in

konny0311 commented:
・gemfile内不要コメントを削除
・変数を外部ファイルから呼び込み
・要素にクラス名付け

hzak2xx commented on app/assets/stylesheets/_config.scss:
```diff
@@ -0,0 +1,26 @@
+// framecolor variables
```
configフォルダをstylesheetsディレクトリ内に作成しに、その中でvariables.scssというファイルを作ってそこに記述してください。

hzak2xx commented on app/assets/stylesheets/_messages.scss:
```diff
@@ -0,0 +1,186 @@
+@import "config", "mixin/mixin", "mixin/notification";
```
importは`application.scss`で行いましょう

hzak2xx commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,65 @@
+    /registered groups shown
+    .contents_main--left
+      %ul.all-groups
+        %li.registered-group Group_name--a
```
こちら、`li.registerd-group`という親要素の中にgroup-nameとgroup-messageという要素を子要素としていれたいですね

Commit:
Created Config/_variables and @import added in application.scss

Commit:
Delete unnecessary function

Commit:
Corrected in index.html.haml

mkoma414 commented on app/views/messages/index.html.haml:
```diff
@@ -13,55 +10,42 @@
-            </i>
-
+            %input#upfile{:accept => "image/*", :name => "upfile", :style => "display:none", :type => "file"}/
+            %i.fa.fa-image{:onclick => "$('#upfile').click()"}
```
アプリケーション開発をする際は、極力コードを役割ごとに分離して記述します。
そのためhtmlの中にjQueryのコードがあるのはあまり好ましくありません。

ただ、このケースで大きな支障はないと思うので、このまま勧めていただいて
大丈夫です。もし余力があれば、もう少しスッキリした書き方もできるので
記事を探したりされるとスキルを磨くことができると思います！
